main.cpp

```diff
@@ -3718,16 +3741,34 @@ bool config_command::execute(device_map &devices) {
 bool info_command::execute(device_map &devices) {
     fos.first_column(0); fos.hanging_indent(0);
     if (!settings.filenames[0].empty()) {
-        auto access = get_file_memory_access(0);
+        uint32_t next_id = 0;
+        auto access = get_file_memory_access(0, false, &next_id);
         uint32_t id = 0;
         id = get_family_id(0);
         if (id == RP2040_FAMILY_ID) {
             access.set_model(rp2040);
         } else if (id >= RP2350_ARM_S_FAMILY_ID && id <= RP2350_ARM_NS_FAMILY_ID) {
             access.set_model(rp2350);
         }
-        fos << "File " << settings.filenames[0] << ":\n\n";
-        info_guts(access, nullptr);
+        if (next_id) {
+            next_id = id;
+            while (next_id) {
+                fos.first_column(0); fos.hanging_indent(0);
+                std::stringstream s;
+                s << "File " << settings.filenames[0] << " family ID " << family_name(next_id) << ":";
+                if (next_id != id) {
+                    string dashes;
+                    std::generate_n(std::back_inserter(dashes), s.str().length() + 1, [] { return '-'; });
+                    fos << "\n" << dashes << "\n";
+                }
+                fos << s.str() << "\n\n";
+                auto tmp_access = get_file_memory_access(0, false, &next_id);
+                info_guts(tmp_access, nullptr);
+            }
+        } else {
+            fos << "File " << settings.filenames[0] << ":\n\n";
```

Comment:
Not directly-related to this change, but is it also worth displaying the family id for chips that _don't_ have a `next_id` ?

Reply:
Added, so it now displays family id for all UF2 files, not just multi-family UF2 files